pkg/generate/app/cmd/newapp.go

```diff
@@ -710,20 +710,29 @@ func (c *AppConfig) Run() (*AppResult, error) {
 		}
 	}
 
-	// Only check circular references for `oc new-build`.
-	if c.ExpectToBuild {
-		err = c.checkCircularReferences(objects)
-		if err != nil {
-			if err, ok := err.(app.CircularOutputReferenceError); ok {
+	err = c.checkCircularReferences(objects)
+	if err != nil {
+		if err, ok := err.(app.CircularOutputReferenceError); ok {
+			if c.ExpectToBuild {
+				// circular reference handling for `oc new-build`.
 				if len(c.To) == 0 {
 					// Output reference was generated, return error.
 					return nil, fmt.Errorf("%v, set a different tag with --to", err)
 				}
 				// Output reference was explicitly provided, print warning.
 				fmt.Fprintf(c.ErrOut, "--> WARNING: %v\n", err)
 			} else {
-				return nil, err
+				// circular reference handling for `oc new-app`
+				if len(templateObjects) > 0 {
```

Comment:
@gabemontero it is possible for us to be generating new objects as well as processing template objects. We should check for circular references on the 2 sets of objects separately if we want different behaviors for each.